report/report.tex

```diff
@@ -26,24 +27,180 @@
+                        
+\begin{minted}{haskell}
+a :: Int;
+b :: Int;
+procedure x() {
+    call y;
+}
+procedure y() {
+    a += 1;
+    b += 1;
+}
+\end{minted}
+
+\noindent This procedure is compiled to something like the following:
+
+\begin{minted}{haskell}
+x a b = do
+    let temp = y a b
+    let (a,b) = temp
+    (a,b)
+y a b = do
+    let temp = a
+    let a = temp + 1
+    let temp = b
+    let b = temp + 1
+    (a, b)
+\end{minted}
+
+\subsubsection{Local Variables}
+The other type of variable is the local variable. Local variables are declared in a \emph{local} statement with a name, type and default value, and undeclared in an \emph{unlocal} statement with a final value. By the time the unlocal statement is being executed, the local variable should have the specified final value. Note however, that the final value can also be a non-literal expression. The unlocal block is there to function as initial value for the local variable if the procedure that it is part of is being uncalled, i.e. reversed: in that case, the local variable needs what as declared as its final value as its initial value, and vice versa. By the time the unlocal block is reached, an assertion is executed to make sure that the procedure can safely be reverted without giving the local variable an incorrect initial value. An unlocal block is implemented as a \inlinecode{let-in} Haskell statement, because this automatically takes care of removing a local variable from scope after the unlocal line. To keep it simple, nesting multiple local blocks with identical variable names is impossible, and local variables can never have the name of a global variable. The let-in block again causes scoping issues, for example here:
```

Comment:
This sentence is not wel formed: the local variable needs what as declared as its final value as its initial value, and vice versa.

Reply:
Fixed